pkgs/tools/package-management/lix/common.nix

```diff
@@ -161,6 +164,9 @@ stdenv.mkDerivation {
 
   postPatch = ''
     patchShebangs --build tests
+  '' + lib.optionalString (withNixDebug != null) ''
+    grep -r -l -Z -e 'env."NIX_LOG_FD".*"2"' src | \
+      xargs -0 sed -i '/NIX_LOG_FD.*2/a env["NIX_DEBUG"] = "${toString withNixDebug}";'
```

Comment:
The idea gets an ACK from me but this snippet must be refactored; it is unreadable.

Ideally, we'd have this as a build-time (or even runtime) feature in Lix.

Reply:
I'm going to collect more feedback before investing the work to do that, especially if the idea itself gets mondo pushback from others. Having it work with upstream support definitely is "the right way", for sure.

Reply:
I would say that the complain of Atemu is nonsense, given that grep and xargs are usually unreadable.

However, to me it looks like you are editing C source code, and in those cases I prefer to use parametric patches.